src/DiagnosticsManager.cpp

```
@@ -0,0 +1,698 @@
+    ConfigUpdate cfg;
+    ConfigurationManager::getInstance().getCurrentConfig(cfg);
+    bool enabled = (cfg.spoolman_on != 0) && (strlen(cfg.spoolman_url) > 0);
+    if (!enabled) {
+        addResult(DiagnosticTest::SPOOLMAN_REACHABILITY, DiagnosticStatus::SKIPPED, 0, 0,
+                  "Spoolman not configured", "");
+        return;
+    }
+    char safeUrl[128];
+    diagRedactUrl(safeUrl, sizeof(safeUrl), cfg.spoolman_url);
+
+    bool held = g_httpMutex && (xSemaphoreTake(g_httpMutex, pdMS_TO_TICKS(DIAG_HTTP_MUTEX_MS)) == pdTRUE);
```

Clang (14.0.6)

[warning] implicit conversion 'SemaphoreHandle_t' (aka 'void *') -> bool

(readability-implicit-bool-conversion)